src/time-picker/TimePicker.js

```diff
@@ -136,13 +139,14 @@ export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
   return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
 };
 
-export const makeOptions = ({ minTime, maxTime, timeFormat }, inputValue) => {
+export const makeOptions = ({ minTime, maxTime, timeFormat, userValue }, inputValue) => {
   const time = parseInTimeFormat(inputValue, timeFormat);
-  const timeList = time === inputError ? [] : createTimeList(time, timeFormat);
+  const selectedValue = (userValue && userValue !== inputError) ? { ...userValue, timeFormat } : userValue;
+  const timeList = (time === inputError ? [] : createTimeList(time, timeFormat)).concat(compact([selectedValue]));
   const filteredTimeList = timeList.filter(filterTime({
-    originalInput: time.originalInput, minTime, maxTime
+    originalInput: time.originalInput || '', minTime, maxTime
```

Comment:
I would edit this function (or add a second `.filter` step) by filtering also `undefined` values to avoid the `.concat(compact([selectedValue]))`.